src/components/SlowPageAlert/index.tsx

```diff
@@ -0,0 +1,43 @@
+import { Location } from 'history';
+import * as React from 'react';
+import { Alert } from 'react-bootstrap';
+
+import {
+  allowSlowPagesParam,
+  createAdjustedQueryString,
+  shouldAllowSlowPages,
+} from '../../utils';
+
+type GetTextForSlowState = (allowSlowPages: boolean) => string;
+
+export type PublicProps = {
+  _shouldAllowSlowPages?: (location: Location) => boolean;
+  getLinkText: GetTextForSlowState;
+  getMessage: GetTextForSlowState;
```

Comment:
I don't think I understand why this component needs those two getters.

Given that it will know about whether to allow/not allow slow pages _via_ `_shouldAllowSlowPages`, why can't the component decide on the messages? I looked at #951 and these messages are static, so I guess it would simplify everything if a `SlowPageAlert` was capable of choosing the right messages.

Reply:
The messages are going to be different for `<CodeView>` -- it won't use the word *diff*.

Reply:
ah, interesting. I still think it's a bit convoluted for what it does but it does not hurt I guess.